jaxrs-tck/src/main/java/jakarta/ws/rs/tck/common/JAXRSCommonClient.java

```diff
@@ -0,0 +1,1212 @@
+
+import java.io.*;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Enumeration;
+import java.util.Hashtable;
+import java.util.Properties;
+import java.nio.charset.StandardCharsets;
+
+
+import org.apache.commons.httpclient.Header;
+import org.apache.commons.httpclient.HttpState;
+
+import jakarta.ws.rs.tck.lib.util.TestUtil;
+import jakarta.ws.rs.tck.common.webclient.TestFailureException;
+import jakarta.ws.rs.tck.common.webclient.WebTestCase;
+import jakarta.ws.rs.tck.common.webclient.http.HttpRequest;
+import jakarta.ws.rs.tck.common.webclient.validation.CheckOneOfStatusesTokenizedValidator;
+import jakarta.ws.rs.tck.common.util.JaxrsUtil;
+import jakarta.ws.rs.tck.common.util.Data;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+
+/**
+ * 
+ * @author dianne jiao
+ * @author jan supol
+ */
+//public abstract class JAXRSCommonClient extends ServiceEETest {
+public abstract class JAXRSCommonClient {
+
+  private static final long serialVersionUID = 1L;
+
+  /**
+   * TS Webserver host property
+   */
+  protected static final String SERVLETHOSTPROP = "webServerHost";
+
+  /**
+   * TS Webserver port property
+   */
+  protected static final String SERVLETPORTPROP = "webServerPort";
+
+  /**
+   * TS home property
+   */
+  protected static final String TSHOME = "ts_home";
+
+  /**
+   * Test properties
+   */
+  protected static final Hashtable<Property, String> TEST_PROPS = new Hashtable<Property, String>();
+
+  /**
+   * StatusCode property
+   */
+  protected static final String STATUS_CODE = "STATUS_CODE";
+
+  /**
+   * Request property
+   */
+  protected static final String REQUEST = "REQUEST";
+
+  /**
+   * Request headers property
+   */
+  protected static final String REQUEST_HEADERS = "REQUEST_HEADERS";
+
+  /**
+   * Goldenfile property
+   */
+  protected static final String GOLDENFILE = "goldenfile";
+
+  /**
+   * Search string property
+   */
+  protected static final String SEARCH_STRING = "SEARCH_STRING";
+
+  /**
+   * Search string case insensitive property
+   */
+  protected String TESTDIR = null;
+
+  /**
+   * Goldenfile directory
+   */
+  protected String GOLDENFILEDIR = "/src/web";
+
+  /**
+   * Default request method
+   */
+  protected static final String GET = "GET ";
+
+  /**
+   * HTTP 1.0
+   */
+  protected static final String HTTP10 = " HTTP/1.0";
+
+  /**
+   * HTTP 1.1
+   */
+  protected static final String HTTP11 = " HTTP/1.1";
+
+  /**
+   * Forward slash
+   */
+  protected static final String SL = "/";
+
+  /**
+   * Goldenfile suffix
+   */
+  protected static final String GF_SUFFIX = ".gf";
+
+  /**
+   * JSP suffix
+   */
+  /**
+   * Current test name
+   */
+  protected String _testName = null;
+
+  /**
+   * location of _tsHome
+   */
+  protected String _tsHome = null;
```

Comment:
Do we expect the golden files in below code to be accessed which seems to depend on _tsHome?

Reply:
The golden files are not used for the current set of tests, it is possible that it is used for pending tests. Shall we refactor the code to remove the occurence of tshome in future changes ?

Reply:
@alwin-joseph  I like your answer that there may be some remaining tests that depend on having a way to obtain the root test folder home (`tshome`) so that certain files (e.g. golden files) can be located within it.  IMO, it is fine to expect `tshome` in future releases but we should standardize how we define system properties that are shared across different TCKs (like `tshome`).  How about creating a text file listing the TCK system properties in the Platform TCK?  We probably could place them in a new folder under https://github.com/eclipse-ee4j/jakartaee-tck (perhaps called design or contracts).  Thoughts?

Reply:
Agree.  System properties should be standardized across TCKs and platform tck repo can be used as place holder for defining the same.